_posts/2023-04-02-writing-custom-pod-security-providers.md

```diff
@@ -0,0 +1,271 @@
+   However, since this example doesn't require any special storage considerations, we can ignore it.
+2. The second type of information is the security context configured by the user directly in the custom resource. 
+   In the Pod Security Providers provided by Strimzi, the user-provided security context always takes priority over whatever the provider sets. 
+   However, it's up to you whether your custom provider decides to respect the user-configured security context or ignore it. 
+   In this example, we will ignore the user-configured security context for simplicity.
+
+For our example with the read-only root filesystem, we do not care about the Pod Security Context because our configuration is part of the container Security Context.
+So we set all the Pod methods to simply return `null`, which means that no Pod Security Context should be set.
+For example:
+
+```java
+    @Override
+    public PodSecurityContext kafkaPodSecurityContext(PodSecurityProviderContext context) {
+        return null;
+    }
+```
+
+And in the container methods, we always return the Security Context to enable the read-only root filesystem:
+
+```java
+    @Override
+    public SecurityContext kafkaContainerSecurityContext(ContainerSecurityProviderContext context) {
+        return new SecurityContextBuilder()
+                .withReadOnlyRootFilesystem()
+                .build();
+    }
+
+    @Override
+    public SecurityContext kafkaInitContainerSecurityContext(ContainerSecurityProviderContext context) {
+        return new SecurityContextBuilder()
+                .withReadOnlyRootFilesystem()
+                .build();
+    }
+```
+
+You can find the full source code in the GitHub repository linked at the end of the blog post.
+
+#### Extending an existing policy
+
+If you want to do only some small changes to an existing provider, you do not need to implement everything from scratch.
+You can just extend the provider and re-implement only the methods you want to change.
+Strimzi currently includes two providers: [`BaselinePodSecurityProvider` and `RestrictedPodSecurityProvider`](https://github.com/strimzi/strimzi-kafka-operator/tree/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl)
+
+For our example, we are going to take the [`RestrictedPodSecurityProvider`](https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl/RestrictedPodSecurityProvider.java).
+It configures the security context of the Strimzi operands to match Kubernetes' _restricted_ security profile.
+When you try to use the Kafka Connect Build and its [Kaniko](https://github.com/GoogleContainerTools/kaniko) builder with this provider, it will [throw an exception](https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl/RestrictedPodSecurityProvider.java#L94-L103) because the Kaniko container builder does not work under the _restricted_ profile.
```

Comment:
you should probably link not to `main` (because if we committed changes to main the line numbers would change make the blog stale), but to some specific tag or commit.